www/components/itinerum_tutorial/Tutorial.jsx

```diff
@@ -1,11 +1,47 @@
 /*
  * src/app/help-panel/Tutorial.jsx
  */
-
 class TutorialComponent extends React.Component {
+    propTypes: {
+        isValid: React.PropTypes.bool,
+        result: React.PropTypes.object,
+        schema: React.PropTypes.object,
+        changeresult: React.PropTypes.func
+    }
+    constructor(props) {
+      super(props);
+      this.state = {
+        result: {
+          fname: this.props.schema.fname,
+          lname: this.props.schema.lname
+        }
+      };
+      this.handlefnameChange = this.handlefnameChange.bind(this);
+      this.handlelnameChange = this.handlelnameChange.bind(this);
+    }
+    handlefnameChange(event) {
+      this.props.changeresult('fname', event.target.value)
```

Comment:
Presumably you would need to change this as well.

```suggestion
      this.props.changeresult(this.state.result)
```